### ISSUE TITLE
Danger situation from losing code if Ctrl+S isn't pressed before exit.

### ISSUE THREAD

**Issue by BDisp** [gui-cs/TerminalGuiDesigner#57]
There is no way to avoid the console closing after user clicked the X icon or press Alt+F4. What I suggesting is always auto saving the changes to a temporary file. If the user unexpected close the console the changes already are in that temporary file. If the user saves the designer before exit that temporary file is deleted. When the user reopen the designer later if still exist the temporary file, the tool will ask to the user if he wants to save the unsaved changes and thus this tool will be more reliability for the user.

**Comment by tznind** [comment]
Good idea.  Also related to this is https://github.com/gui-cs/TerminalGuiDesigner/issues/36 

I think first step is to track unsaved changes and warn when closing with Ctrl+Q.  Then this is a logical next step to offer recovery like you say.

**Comment by tznind** [comment]
There is now a warning when trying to exit with ctrl+Q so this is less important but still a nice feature to have

**Comment by tznind** [comment]
Yup, sorry.  That comment was just for anyone comming across this issue later to say that its planned but not urgent.
